### PR TITLE
wip - rust - templates and scripts - v1

### PR DESCRIPTION
**Pull request by jasonish**
Changes:
- C stub app-layer-template-rust for Rust based parsers.
- C stub output-json-template-rust for Rust based parsers.
- Replace setup-app-layer.sh and setup-app-layer-logger.sh with setup-app-layer.py.

And the start of an app-layer written in Rust. I was thinking of a protocol that had length prefixed messages like "\x0cHello World!". But given we want to turn this into Gopher, maybe something like "12:Hello World!" would be better?

Prscript:
- PR jasonish-pcap: https://buildbot.openinfosecfoundation.org/builders/jasonish-pcap/builds/321
- PR jasonish: https://buildbot.openinfosecfoundation.org/builders/jasonish/builds/674

### CODE REVIEW

**Comment by victorjulien** [comment]
<pre>
./scripts/setup-app-layer.py --rust MYPROTO
Generating src/app-layer-myproto.c.
Generating src/app-layer-myproto.h.
Patching src/Makefile.am.
Patching src/app-layer-protos.h.
Patching src/app-layer-protos.c.
Patching src/app-layer-detect-proto.c.
Patching src/app-layer-parser.c.
Patching suricata.yaml.in.
Generating src/output-json-myproto.h.
Generating src/output-json-myproto.c.
Patching src/Makefile.am.
Patching src/suricata-common.h.
Patching src/output.c.
Patching suricata.yaml.in.
Patching src/util-profiling.c.

An application detector and parser for the protocol MYPROTO has
now been setup in the files:

    src/app-layer-myproto.h
    src/app-layer-myproto.c

A JSON application layer transaction logger for the protocol
MYPROTO has now been set in the files:

    src/output-json-myproto.h
    src/output-json-myproto.c

Suricata should now build cleanly. Try running "make".
</pre>

Am I doing something wrong wrt --rust?

File: scripts/setup-app-layer.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (victorjulien) [inline scripts/setup-app-layer.py]: lots of newlines

**Comment by jasonish** [comment]
Rust part isn't complete yet. It should still build though, just not call
out to the register. Ahh, the messages should also change to point to Rust
source code.

**Comment by jasonish** [comment]
Closing. See https://github.com/OISF/suricata/pull/3484

**Pull request closed without merging**